File: r_code/data_prep/data_generation.r

############################
## Sourcing the functions ##
############################

source("r_code/data_prep/data_gen_functions.R")
source("r_code/utils/utils.R")

#################################
## Defining output directories ##
#################################

## Main output directory:
OUTPUT_DIR <- file.path("..")

## Generated data:
DATA_DIR <- file.path(OUTPUT_DIR, "data")
create_directory(DATA_DIR)

#########################
## Simulating the data ##
#########################

## Creating a simulated count matrix and associated metadata
counts_metadata_list <- generate_counts_with_metadata(
    output_path = DATA_DIR,
    number_of_genes = 1000,
    number_of_samples = 10,
    seed = 567,
    condition_vector = c("Control", "Treatment"))

File: python_code/data_prep/data_generation.py
import os

from data_gen_functions import *
from pathlib import Path

#########################
## Simulating the data ##
#########################

## Main output directory:
OUTPUT_DIR = Path('results')
os.makedirs(OUTPUT_DIR, exist_ok=True)

## Generated data:
DATA_DIR = os.path.join(OUTPUT_DIR, 'data')
os.makedirs(DATA_DIR, exist_ok=True)

COUNT_MAT_PATH = os.path.join(DATA_DIR, 'count_matrix.pkl')
METADATA_PATH = os.path.join(DATA_DIR, 'metadata.tsv')

## Create example RNA-seq count matrix and metadata
count_df,metadata = generate_counts_with_metadata(
    COUNT_MAT_PATH,
    METADATA_PATH,
    number_of_genes=1000,
    number_of_samples=10,
    condition_vector=['Control', 'Treatment'],
    seed=567,
)